packages/dds/sequence/src/sequenceDeltaEvent.ts

```diff
@@ -19,22 +20,28 @@ import {
 /**
  * Base class for SequenceDeltaEvent and SequenceMaintenanceEvent.
  *
- * The properties of this object and its sub-objects represent a point in time state
- * at the time the operation was applied. They will not take into any future modifications
- * performed to the underlying sequence and merge tree.
+ * The properties of this object and its sub-objects represent the state of the sequence at the
+ * point in time at which the operation was applied.
+ * They will not take into any future modifications performed to the underlying sequence and merge tree.
  */
 export abstract class SequenceEvent<TOperation extends MergeTreeDeltaOperationTypes = MergeTreeDeltaOperationTypes> {
+    /**
+     * @deprecated - Events no longer fire when the change they correspond to had no impact (e.g. a remote delete
+     * event for a range that had already been deleted locally).
+     * Clients can therefore assume this property is false.
+     */
     public readonly isEmpty: boolean;
     public readonly deltaOperation: TOperation;
     private readonly sortedRanges: Lazy<SortedSegmentSet<ISequenceDeltaRange<TOperation>>>;
-    private readonly pFirst: Lazy<ISequenceDeltaRange<TOperation> | undefined>;
-    private readonly pLast: Lazy<ISequenceDeltaRange<TOperation> | undefined>;
+    private readonly pFirst: Lazy<ISequenceDeltaRange<TOperation>>;
+    private readonly pLast: Lazy<ISequenceDeltaRange<TOperation>>;
 
     constructor(
         public readonly deltaArgs: IMergeTreeDeltaCallbackArgs<TOperation>,
         private readonly mergeTreeClient: Client,
     ) {
-        this.isEmpty = deltaArgs.deltaSegments.length === 0;
+        assert(deltaArgs.deltaSegments.length > 0, 0x2d8 /* 'Empty change event should not be emitted.' */);
```

Comment:
where does that short code come from? in general we check in with strings, and covert to short codes only before release

Reply:
ahh, That makes sense. I was wondering how the error code generation strategy works :). I generated it by running policy-check after saving the file with the string version. I'll update the comment on `assert` to make the intended process more clear.

Reply:
though, still debugging to figure out why this assert fails on some annotation ops (fix is almost certainly to just to the same limiting I've done on delete ops there, but I want to figure out what scenario causes it first)

Reply:
yeah. looks like its related to snapshot loading